fedbadges/utils.py

```diff
@@ -111,13 +112,30 @@ def notification_callback(topic, msg):
 def user_exists_in_fas(config, user):
     """ Return true if the user exists in FAS. """
     default_url = 'https://admin.fedoraproject.org/accounts/'
+    if config.get["fasjson"]:
+        return user_exists_in_freeipa_fas(config, user)
+
     fas2 = fedora.client.AccountSystem(
         base_url=config['fas_credentials'].get('base_url', default_url),
         username=config['fas_credentials']['username'],
         password=config['fas_credentials']['password'],
     )
     return bool(fas2.person_by_username(user))
 
+def user_exists_in_fasjson(config, user):
+    """ Return true if the user exists in freeipa-fas. """
+    default_url = 'https://admin.fedoraproject.org/accounts/'
+    client = Client(
+        url=config['fas_credentials'].get('base_url', default_url)
+    )
+    try:
+        client.get_user(username=user).result
+        return True
+    except fasjson_client.errors.APIError as e:
+        log.exception(e)
```

Comment:
We should probably log the exception only if it's unexpected. Since we're checking whether the user exists, let's look at the APIError and only log if the `code` attribute is not 404.